xla/mlir/backends/gpu/transforms/lmhlo_gpu_to_gpu_runtime.cc

```diff
@@ -112,11 +112,14 @@ class GemmOpLowering : public OpRewritePattern<GEMMOp> {
     call->setAttr(b.getStringAttr("uid"), b.getI64IntegerAttr(uid_.uid()));
 
     // Copy backend specific attributes.
-    auto algorithm_attr =
-        op.getAlgorithm()
-            ? op.getAlgorithmAttr()
+    auto algorithm_attr = op.getAlgorithm() ? op.getAlgorithmAttr()
             : b.getI64IntegerAttr(stream_executor::blas::kDefaultGemmAlgo);
     call->setAttr(b.getStringAttr("algorithm"), algorithm_attr);
+    call->setAttr(b.getStringAttr("grad_x"), 
+          op.getGradX() ? op.getGradXAttr() : b.getBoolAttr(false));
+    call->setAttr(b.getStringAttr("grad_y"), 
+          op.getGradY() ? op.getGradYAttr() : b.getBoolAttr(false));
```

Comment:
Why set this attribute at all, when op.getGradY() is "false"?

(Just asking)

Reply:
yes, although grad_x/y attributes are optional (same as **algorithm** attribute above), some default value should be set if the attribute is not present, otherwise the gemm [custom call](https://github.com/ROCmSoftwarePlatform/xla/blob/ci_grad_xy_propagation/xla/service/gpu/runtime/gpublas_lt_matmul.cc#L241-L242) would fail, with the error like:

```
Actual: false (INTERNAL: Failed to execute XLA Runtime executable: run time error: custom call 'xla.gpu.cublas.lt.matmul' failed: Wrong number of attributes: expected 10 got 8; current tracing scope: custom-call; current profiling annotation: XlaModule:#hlo_module=MatMul.10,program_id=13#.)
```

Reply:
ok thanks